routing/routing_session.cpp

```diff
@@ -382,6 +382,7 @@ void RoutingSession::PassCheckpoints()
 {
   while (!m_checkpoints.IsFinished() && m_route->IsSubroutePassed(m_checkpoints.GetPassedIdx()))
   {
+    m_route->PassNextSubroute();
     m_checkpoints.PassNextPoint();
```

Comment:
Yes. RoutingSession provides info about checkpoints to Route, Route manages m_poly. m_route->PassNextSubroute() call will do the job for Route and m_poly inside it.